src/main/java/software/amazon/awssdk/eventstreamrpc/RpcServer.java

```diff
@@ -67,33 +76,37 @@ public RpcServer(EventLoopGroup eventLoopGroup, SocketOptions socketOptions, Tls
     public void runServer() {
         validateServiceHandler();
         if (!serverRunning.compareAndSet(false, true)) {
-            throw new IllegalStateException("Failed to start IpcServer. It's already started or has not completed a prior shutdown!");
+            throw new IllegalStateException(
+                    "Failed to start IpcServer. It's already started or has not completed a prior shutdown!");
         }
         serverBootstrap = new ServerBootstrap(eventLoopGroup);
         tlsContext = tlsContextOptions != null ? new ServerTlsContext(tlsContextOptions) : null;
-        listener = new ServerListener(hostname, port, socketOptions, tlsContext, serverBootstrap, new ServerListenerHandler() {
-                @Override
-                public ServerConnectionHandler onNewConnection(ServerConnection serverConnection, int errorCode) {
-                    if (serverConnection == null) {
-                        LOGGER.info("New connection immediately closed");
-                        return null;
-                    }
-                    try {
-                        LOGGER.info("New connection code [" + CRT.awsErrorName(errorCode) + "] for " + serverConnection.getResourceLogDescription());
-                        final ServiceOperationMappingContinuationHandler operationHandler =
-                                new ServiceOperationMappingContinuationHandler(serverConnection, eventStreamRPCServiceHandler);
-                        return operationHandler;
-                    } catch (Throwable e) {
-                        LOGGER.error("Throwable caught in new connection: " + e.getMessage(), e);
-                        return null;
+        listener = new ServerListener(hostname, port, socketOptions, tlsContext, serverBootstrap,
+                new ServerListenerHandler() {
+                    @Override
+                    public ServerConnectionHandler onNewConnection(ServerConnection serverConnection, int errorCode) {
+                        if (serverConnection == null) {
+                            LOGGER.info("New connection immediately closed");
+                            return null;
+                        }
+                        try {
+                            LOGGER.info("New connection code [" + CRT.awsErrorName(errorCode) + "] for "
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against a allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html